documentation/guide-jpa.asciidoc

```diff
@@ -114,7 +114,7 @@ Using the Aliases API of QueryDSL might result in an `InvalidDataAccessApiUsageE
 . Make sure, that all your custom types used in entities provide a non-argument constructor with at least visibility level `protected`.
 
 === Primary Keys
-We only use simple Long values as primary keys (IDs). By default it is auto generated (`@GeneratedValue(strategy=GenerationType.AUTO)`). This is already provided by the class `com.devonfw.<projectName>.general.dataaccess.api.AbstractPersistenceEntity` that you can extend.
+We only use simple Long values as primary keys (IDs). By default it is auto generated (`@GeneratedValue(strategy=GenerationType.AUTO)`). For spring appliations this is already provided by the class `com.devonfw.<projectName>.general.dataaccess.api.AbstractPersistenceEntity` that you can extend. In Quarkus applications, you need to add the ID attribute to your class manually (see https://github.com/devonfw-sample/devon4quarkus-reference/blob/master/src/main/java/com/devonfw/demoquarkus/domain/model/Animal.java[here]).
```

Comment:
This is just a best practice that has nothing to do with spring or quarkus.
Actually we also need an abstract mapped super-class for this also in quarkus.
I just created [#13](https://github.com/devonfw-sample/devon4quarkus-reference/issues/13) for this.

Reply:
Added therefore the `AbstractPersistenceEntity` to the [modern project structure](https://github.com/GuentherJulian/devon4j/blob/jpa_guide/documentation/guide-structure-modern.asciidoc) guide. Should I mention that this class has to be created manually for the Quarkus application, since we don't have a template for it right now? Or should I leave it as it is?